### PR TITLE
Don't offer to convert to pattern matching when type is nullable.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://developercommunity.visualstudio.com/content/problem/49712/ide0020-problem.html

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @Pilchie

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz 

**Customer scenario**

Our code fix that offers to use pattern-matching produces bogus code in a type is Nullable.

**Bugs this fixes:**

https://developercommunity.visualstudio.com/content/problem/49712/ide0020-problem.html

**Workarounds, if any**

None.

**Risk**

Low.

**Performance impact**

Low.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

Hard to test features against all possible language cases.

**How was the bug found?**

Customer reported.

File: src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpIsAndCastCheckTests.cs [hunk 1]
```diff
@@ -518,5 +518,25 @@ static void N(ParameterSyntax parameter)
...
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestMissingWithNullableType()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"
+class C
+{
+    public object Convert(object value)
+    {
+        if (value is bool?)
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpIsAndCastCheckTests.cs]: :memo: In almost all cases, it's safe to rewrite this as the following:

```csharp
if (value is bool tmp)
{
}
```

The one exception I can think of is a case where the static type of `tmp` is used as a generic argument.

**Comment by CyrusNajmabadi** [comment]
@sharwell We can def consider expanding on what is matched and what we convert to.

**Pull request merged**